arch/arm/core/aarch64/fatal.c

```diff
@@ -168,7 +179,18 @@ static bool is_recoverable(z_arch_esf_t *esf, uint64_t esr, uint64_t far,
 	if (!esf)
 		return false;
 
-	/* Empty */
+#ifdef CONFIG_USERSPACE
+	for (int i = 0; i < ARRAY_SIZE(exceptions); i++) {
```

Comment:
Interesting. This is a very different approach than the one I'm used to.

(I'm used to instead having a spinlock-like approach, where before doing something that may abort you record in thread context that an exception is allowed and the handler thereof, and after you remove said record.)

One word of warning: this approach does not play well with anything that splits / inlines/ duplicates functions. Should be fine here currently because everything is assembly, but worth noting.